### PR TITLE
SubnetCalc: Minor tidying up of subs

### PR DESCRIPTION
**Pull request by mintsoft**
I did this as a test, might as well get it in master

----------
IA Page : https://duck.co/ia/view/subnet_calc

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 1]
```diff
@@ -128,4 +111,21 @@ handle query => sub {
         };
 };
 
-1;
+# Convert an integer into an IP address.
+sub int_to_str {
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: Hmm... Personally I prefer the handler to go at the bottom (just makes more 'flow' sense to me), what do you think?

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 2]
```diff
@@ -13,17 +13,6 @@ zci answer_type => "subnet_calc";
 zci is_cached => 1;
 
 handle query => sub {
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: Hmm... Personally I prefer the handler to go at the bottom (just makes more 'flow' sense to me), what do you think?

Comment (mintsoft) [inline lib/DDG/Goodie/SubnetCalc.pm]: Hmm, I always wax and wane on that, recently I've been finding it easier to debug/review/understand if IA's follow a pattern of:
* usings
* statics
* triggers
* handler
* misc subs
as I usually like to start with the handler, so I want that near the top, as close to the triggers as possible. Technically it makes no difference obviously, Perl will let you define things wherever you like. @MrChrisW what are you preferences here?

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: @mintsoft Yeah - same for me but with `handler` and `misc subs` switched over ;) Though usually I'm not too fussed about where the `triggers` are depending on the code - usually near the top or near the handle though.

Comment (MrChrisW) [inline lib/DDG/Goodie/SubnetCalc.pm]: @mintsoft My preference is to have the subs below the handler. If I'm debugging I'll look at the handler first then follow the program flow from there (moving to other subs) - in this case it makes sense to have handler first. It really is just personal preference though! :+1:

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: @MrChrisW @mintsoft I'm outnumbered!

Very well, this isn't a *big* issue for me - as long as it is split into reasonable chunks!

File: lib/DDG/Goodie/SubnetCalc.pm (hunk 1 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: Why `int_to_str` and not `int_to_ip_str`?

File: lib/DDG/Goodie/SubnetCalc.pm (hunk 2 above)
Comment (MrChrisW) [inline lib/DDG/Goodie/SubnetCalc.pm]: > as long as it is split into reasonable chunks!
@GuiltyDolphin Agreed! :+1:

File: lib/DDG/Goodie/SubnetCalc.pm (hunk 1 above)
Comment (mintsoft) [inline lib/DDG/Goodie/SubnetCalc.pm]: :+1:

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 3]
```diff
@@ -128,4 +111,21 @@ handle query => sub {
...
-1;
+# Convert an integer into an IP address.
+sub int_to_str {
+    my ($ip) = @_;
+    sprintf "%u.%u.%u.%u", $ip >> 24 & 0xff, $ip >> 16 & 0xff, $ip >> 8 & 0xff, $ip & 0xff;
+}
+
+# Convert an IP address into an integer.
+sub ip_to_int {
+    (int($_[0]) << 24) + (int($_[1]) << 16) + (int($_[2]) << 8) + int($_[3]);
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: Lot of bit shifting going on in this file :wink:

**Comment by GuiltyDolphin** [comment]
@mintsoft The trigger regex seems to have a lot of repetition in it - could we condense it a bit to make it easier to read?

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 4]
```diff
@@ -128,4 +111,21 @@ handle query => sub {
         };
 };
 
-1;
+# Convert an integer into an IP address.
+sub int_to_ip_str {
+    my ($ip) = @_;
+    sprintf "%u.%u.%u.%u", $ip >> 24 & 0xff, $ip >> 16 & 0xff, $ip >> 8 & 0xff, $ip & 0xff;
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: Seeing as the second half of this is the inverse of `ip_bytes_to_int`, perhaps we could pull it out into `int_to_ip_bytes` to make it clearer?

Comment (mintsoft) [inline lib/DDG/Goodie/SubnetCalc.pm]: Hmm, I think personally I'd rather not, if it was used in multiple places, then I'd say yes. This is relatively standard for anything dealing with IP addresses, the ipv4 format is well documented etc. There's no value that I can see to provide an explicit function that returns an array other than maintaining symmetry of the subs; which seems kinda pointless to me

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/SubnetCalc.pm]: @mintsoft Fair enough.

**Comment by GuiltyDolphin** [comment]
@mintsoft This is looking pretty good!

Much clearer than before - let me know when you're happy.

**Comment by mintsoft** [comment]
btw @GuiltyDolphin I'm happy with this if you are

**Comment by GuiltyDolphin** [comment]
@mintsoft Sweet! I've gone through it again and it looks much better! :dolphin:

**Pull request merged**